### ISSUE TITLE
[Apps & Marketplace V2] Add RKE1 Linux tolerations and RKE2 nodeSelectors to all feature charts

### ISSUE THREAD

**Issue by aiyengar2** [rancher/rancher#36403]
Child Issues:
- https://github.com/rancher/rancher/issues/36265
- https://github.com/rancher/rancher/issues/36266

#### Context

In RKE1 clusters:
- Linux nodes are **unmarked**

As a result, designing a Helm chart that would just work for this cluster **does not require nodeSelectors or tolerations**.

In RKE1 Windows clusters:
- Windows nodes are **unmarked**
- Linux nodes are **tainted**

As a result, designing a Helm chart that would just work for this cluster **requires Linux workloads to tolerate our taint**.

In RKE2 clusters:
- Windows nodes are **labeled** based on OS annotations
- Linux nodes are **labeled** based on OS annotations

As a result, designing a Helm chart that would just work for this cluster **requires Linux workloads to use nodeSelectors AND requires Windows workloads to use nodeSelectors**.

#### Required Change

Since RKE2 clusters require nodeSelectors to be specified, **all of our feature charts need to implement those nodeSelectors**.

In addition, if we are adding nodeSelectors to all of our charts to support RKE2 Windows, we might as well add the tolerations to support RKE1 Windows as well. So both of these can be done together.

Therefore, once we add these changes, we can then update the required annotations from https://github.com/rancher/rancher/issues/36401.

#### Solution

We effectively need to add the following helper templates onto every Linux workload on every one of our charts: https://github.com/rancher/charts/blob/502035b9ac9763e2ebbdde5c05f2137862bb2678/charts/rancher-monitoring/100.1.0%2Bup19.0.3/templates/_helpers.tpl#L106-L124

Adding the `linux-node-tolerations` template ensures that we deploy Linux workloads onto **Linux nodes only** on RKE1 Windows clusters.

Adding the `linux-node-selector` template ensures that we deploy Linux workloads onto **Linux nodes only** on RKE2 clusters.

Once these are added, based on https://github.com/rancher/rancher/issues/36401 we should also add in `catalog.cattle.io/permits-os: linux,windows` to all of our charts.

**Comment by sirredbeard** [comment]
Awaiting updates from Arvind and PR https://github.com/rancher/charts/pull/1813

**Comment by sirredbeard** [comment]
Unblocked. Pending dev validation by Team 3.

**Comment by PennyScissors** [comment]
Closing since Zube bot mistakenly re-opened this and it is superseded by a previous dashboard issue.
https://github.com/rancher/rancher/issues/36403#issuecomment-1062374848
